### PR TITLE
[INFRA-1193] Query Artifactory for LTS baselines

### PR DESCRIPTION
**Pull request by daniel-beck**
@olivergondza FYI

@rtyler PTAL

### CODE REVIEW

File: scripts/generate-javadoc.sh [hunk 1]
```diff
@@ -33,11 +33,19 @@ function generate_javadoc_core() {
...
-for release in 1.554 1.565 1.580 1.596 1.609 1.625 1.642 1.651 2.7 2.19 2.32 2.46 2.60; do
-    echo ">> Found release ${release}"
-    generate_javadoc_core "${release}"
+
+wget -O jq https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64 || { echo "Failed to download jq" >&2 ; exit 1; }
+chmod +x jq || { echo "Failed to make jq executable" >&2 ; exit 1; }
+
+set -o pipefail
+
+RELEASES=$( curl 'https://repo.jenkins-ci.org/api/search/versions?g=org.jenkins-ci.main&a=jenkins-core&repos=releases&v=?.*.1' | ./jq --raw-output '.results[].version' | head -n 10 )
```
Comment (daniel-beck) [inline scripts/generate-javadoc.sh]: Needs `||` otherwise `pipefail` doesn't do anything.

**Comment by rtyler** [comment]
sure I guess *shrug*

**Comment by olivergondza** [comment]
LGTM, though I do not understand the comment, Daniel.

**Comment by daniel-beck** [comment]
> I do not understand the comment

I `set -o pipefail` which makes a sequence of pipe connected commands not return the rightmost exit code (default), but the rightmost non-zero exit code. This ensures no errors are discarded by pipes.

However, `errexit` isn't set, and I don't use `command || { echo 'command failed' >&2 ; exit 1 ; }` which `pipefail` doesn't effectively change anything -- the exit codes are unused.

**Comment by daniel-beck** [comment]
IMO done now.

**Pull request merged**